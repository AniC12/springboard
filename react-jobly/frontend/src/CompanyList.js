import React, { useState, useEffect } from "react";
import SearchForm from "./SearchForm";
import JoblyApi from "./api";
import CompanyCard from "./CompanyCard";


function CompanyList() {
  const [companies, setCompanies] = useState([]);

  useEffect(function getCompaniesOnMount() {
    search();
  }, []);

  async function search(name) {
    let companies = await JoblyApi.getCompanies(name);
    setCompanies(companies);
  }

  if (!companies) {
    return <p>Loading &hellip;</p>;
  }

  return (
    <div>
      <SearchForm searchFor={search} />
      {companies.length > 0
        ? (
          <div className="CompanyList-list">
            {companies.map(c => (
              <CompanyCard
                key={c.handle}
                handle={c.handle}
                name={c.name}
                description={c.description}
                logoUrl={c.logoUrl}
              />
            ))}
          </div>
        ) : (
          <p className="lead">Sorry, no results were found!</p>
        )}
    </div>
  );
}

export default CompanyList;