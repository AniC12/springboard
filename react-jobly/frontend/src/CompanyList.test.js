import '@testing-library/jest-dom';
import React from "react";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import CompanyList from "./CompanyList";
import JoblyApi from "./api";

// Mock the JoblyApi.getCompanies method
jest.mock("./api", () => ({
    getCompanies: jest.fn()
}));

it("renders without crashing", () => {
    render(<CompanyList />);
});

it("matches snapshot", function () {
    const { asFragment } = render(<CompanyList />);
    expect(asFragment()).toMatchSnapshot();
});


it("displays companies after API call", async () => {
    const mockCompanies = [
        { handle: "test1", name: "Test Company 1", description: "Desc 1", logoUrl: "http://test1.jpg" },
        { handle: "test2", name: "Test Company 2", description: "Desc 2", logoUrl: "http://test2.jpg" }
    ];
    JoblyApi.getCompanies.mockResolvedValueOnce(mockCompanies);

    render(
        <MemoryRouter>
            <CompanyList />
        </MemoryRouter>
    );

    expect(await screen.findByText('Test Company 1')).toBeInTheDocument();
    expect(await screen.findByText('Test Company 2')).toBeInTheDocument();
});