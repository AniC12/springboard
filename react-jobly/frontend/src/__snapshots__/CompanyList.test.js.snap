// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div>
      <form>
        <input
          name="searchTerm"
          placeholder="Enter search term.."
          value=""
        />
        <button
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
    <p
      class="lead"
    >
      Sorry, no results were found!
    </p>
  </div>
</DocumentFragment>
`;
